test/functional/test_framework/test_node.py

```diff
@@ -645,10 +645,11 @@ def add_p2p_connection(self, p2p_conn, *, wait_for_verack=True, **kwargs):
             p2p_conn.sync_with_ping()
 
             # Consistency check that the node received our user agent string.
-            # Find our connection in getpeerinfo by our address:port, as it is unique.
+            # Find our connection in getpeerinfo by our address:port and theirs, as this combination is unique.
             sockname = p2p_conn._transport.get_extra_info("socket").getsockname()
             our_addr_and_port = f"{sockname[0]}:{sockname[1]}"
-            info = [peer for peer in self.getpeerinfo() if peer["addr"] == our_addr_and_port]
+            dst_addr_and_port =f"{p2p_conn.dstaddr}:{p2p_conn.dstport}"
```

Comment:
nit, missing white space after `=`:
```suggestion
            dst_addr_and_port = f"{p2p_conn.dstaddr}:{p2p_conn.dstport}"
```

Reply:
fixed